Update plans grid to use new blue variables

lsl opened the pull request:
Related to https://github.com/Automattic/wp-calypso/issues/97537
Related to https://github.com/Automattic/wp-calypso/issues/97582

## Proposed Changes

* Updates the free/personal/premium blue button colors to match the button colors used elsewhere in onboarding. These are also matching the modern theme which is given to new sites.
* Updates the business / ecommerce purple button colors to WooCommerce color variables found in https://github.com/Automattic/color-studio/blob/trunk/dist/color-properties.css (@nuriapenya please see screenshot below as these purples will update to a brighter color once calypso updates its version of the color-studio package - there won't be any warning when this happens so please review now in case you want me to hardcode things to avoid this change)
* Uses darker background colors on hover instead of opacity to lighten the button, this semi-matches button behavior elsewhere.

## Testing Instructions

* Plans grid shown in onboarding flows e.g. http://calypso.localhost:3000/setup/onboarding/plans?categories=blog&theme=hola
* Plans grid shown in the plans screen e.g. http://calypso.localhost:3000/plans/testingcalypso136.wordpress.com


| Plans step Before  | After |
| ------------- | ------------- |
| ![Screenshot 2024-12-18 at 15-55-13 Create a site — WordPress com](https://github.com/user-attachments/assets/dfaf1f52-d12e-41bc-9702-5d44e9354130) | ![Screenshot 2024-12-18 at 15-55-34 Create a site — WordPress com](https://github.com/user-attachments/assets/498e8f21-7bc2-45dc-9c6d-daac6432fb5b) |

| Plans screen Before (modern theme, new default) | After |
| ------------- | ------------- |
| ![Screenshot 2024-12-18 at 15-56-50 Plans ‹ Site Title — WordPress com](https://github.com/user-attachments/assets/86e99bd4-1a84-4cd7-9497-0faeecc6d69c) | ![Screenshot 2024-12-18 at 15-57-03 Plans ‹ Site Title — WordPress com](https://github.com/user-attachments/assets/a6d33e4d-7640-458b-a535-16967edae128) |

| Plans screen Before ("default" theme) | After |
| ------------- | ------------- |
| ![Screenshot 2024-12-18 at 15-56-17 Plans ‹ Site Title — WordPress com](https://github.com/user-attachments/assets/c5132458-abdc-408e-8b99-0f2691715314) | ![Screenshot 2024-12-18 at 15-56-06 Plans ‹ Site Title — WordPress com](https://github.com/user-attachments/assets/2ad540aa-d314-487e-a664-a33334ad570e) |

| Brighter Woo Purples once color-studio is updated | 
| ------------- |
| ![Screenshot 2024-12-18 at 16-12-56 Create a site — WordPress com](https://github.com/user-attachments/assets/a159e415-90e0-449c-bab8-2a171b597d5c) |

lsl committed:
Update plans grid to use new blue variables

lsl commented:
@southp fyi this PR is more of a proposal than something we need to push right now, it's coming from a slack conversation here: p1734432359203159/1734430127.673289-slack-C085HCWCEDN - Let me know if there's any concerns about these changes. Also curious how to propagate these changes to the LOHP versions like https://wordpress.com/pricing/

@nuriapenya could you take a look at the calypso live links above for onboarding / plans pages and let me know if this is what you were expecting?

arthur791004 reviewed (approved):
Looks good 👍

nuriapenya reviewed (approved):
LGTM!! 💎 thanks for working on these!

> Updates the business / ecommerce purple button colors to WooCommerce color variables found in https://github.com/Automattic/color-studio/blob/trunk/dist/color-properties.css

Cool! Works for me. Will this also be applied to the other grids? (e.g. plans page)

> Also curious how to propagate these changes to the LOHP versions like https://wordpress.com/pricing/

Yes. If possible we should translate these changes everywhere.

lsl committed:
Merge branch 'trunk' into update/plans-blues

lsl committed:
Plan comparison grid checkmark blue

lsl committed:
Specificity fix for deprecated Button / .button conflict

lsl committed:
Switch button text color to --color-text-inverted (--studio-white)

lsl committed:
Merge branch 'trunk' into update/plans-blues

lsl commented:
> It feels like Personal, Business, etc. should have their own variable names in the color-studio package. Using a woocommerce variable for the Business plan feels weird since I naively would associate it with the Commerce plan.

Possibly, I think the plans grid would be the only place that uses it, if we're leveraging woo and wp colors for these I think it's ok to leverage them but I agree it would be a bit nicer.

Updates:
- Fixed the grey text specificity issue in c3237f5673bb48f9e414f8044e4b6448303c48c4
- Moved over to using `--studio-white` text for buttons in 9b271c4a6e83ac85d69c5520c308c948900de1bd

@nuriapenya I noticed font weights were a bit off too, both in the plans and domain steps, proposing a separate fix for that in https://github.com/Automattic/wp-calypso/pull/97631

southp commented:
> @southp fyi this PR is more of a proposal than something we need to push right now, it's coming from a slack conversation here: p1734432359203159/1734430127.673289-slack-C085HCWCEDN - Let me know if there's any concerns about these changes. Also curious how to propagate these changes to the LOHP versions like https://wordpress.com/pricing/

There is currently no direct way to propagate changes to the landing pages. You'd need to implement the same change on Landpack: PCYsg-uKY-p2.

southp commented on packages/plans-grid-next/src/components/comparison-grid/index.tsx:
```diff
@@ -662,7 +662,7 @@ const ComparisonGridFeatureGroupRowCell: React.FunctionComponent< {
 								</span>
 							) }
-							{ hasFeature && ! featureLabel && <Gridicon icon="checkmark" color="#0675C4" /> }
+							{ hasFeature && ! featureLabel && <Gridicon icon="checkmark" color="#3858e9" /> }
```
Non-blocking note: by a quick grep, this color code has been used extensively across the codebase. We should consider to standardize it.

lsl replied:
Good idea, updated to `var(--studio-wordpress-blue-50)`.

southp reviewed (approved):
Thanks for working on making the styles consistent. I've checked the main onboarding flow, the logged-in /plans, some Stepper flows like /setup/newsletter, and setting the locale to non-en just in case. They all worked as expected :)

lsl committed:
Use studio-wordpress-blue-50 variable for gridicon color

lsl committed:
Merge branch 'trunk' into update/plans-blues

lsl committed:
Merge branch 'trunk' into update/plans-blues

lsl commented on packages/plans-grid-next/src/components/plan-button/style.scss:
```diff
@@ -1,62 +1,75 @@
 @import "@automattic/onboarding/styles/mixins";
 
-.plan-features-2023-grid__actions-button {
+.button.plan-features-2023-grid__actions-button {
```
This is causing a regression in the start with free link modal:

![Screenshot(191)](https://github.com/user-attachments/assets/61efb10e-fda9-4615-8d59-558c052d2d27)

lsl replied:
Fixed by 11862ce7f919c0cd7634fb3d4e5d550ce82caaf9

lsl committed:
Remove flex: 1 to prevent conflict in modal

candy02058912 reviewed (commented):
Coming from p1734570117377399/1734514741.812639-slack-C085HCWCEDN

Can confirm the modal buttons are fixed.

![review@2x](https://github.com/user-attachments/assets/19c77bb9-2c26-4233-a1bb-ebdc67878844)

lsl commented:
> It feels like Personal, Business, etc. should have their own variable names in the color-studio package. Using a woocommerce variable for the Business plan feels weird since I naively would associate it with the Commerce plan.

@p-jackson p9Jlb4-fwB-p2 :smile:

p-jackson commented:
> @p-jackson I'm guessing this will change somewhat soon p9Jlb4-fwB-p2 😄

Ch-ch-ch-changes! Blue everywhere! I appreciate the commitment to standard components though 👍
